sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -293,13 +283,35 @@ public boolean isDisposed() {
 
     @Override
     public void dispose() {
+        dispose(null);
+    }
+
+    /**
+     * Disposes of the sender when an exception is encountered.
+     *
+     * @param errorCondition Error condition associated with close operation.
+     */
+    public void dispose(ErrorCondition errorCondition) {
```

Comment:
This dispose was if we had to close the connection because of an error. :/ I expected the `dispose()` to be used when it is closed normally. Let me update this to package-private.